pkgs/data/misc/sing-geosite/default.nix

```diff
@@ -0,0 +1,63 @@
+{ lib
+, buildGoModule
+, fetchFromGitHub
+, v2ray-domain-list-community
+}:
+
+buildGoModule {
+  pname = "sing-geosite";
+  inherit (v2ray-domain-list-community) version;
+
+  src = fetchFromGitHub {
+    owner = "SagerNet";
+    repo = "sing-geosite";
+    rev = "4a32d56c1705f77668beb5828df0b0a051efdeb9";
+    hash = "sha256-P/EBcwJI2G9327BNi84R+q6BABx9DEKpN6ETTp8Q4NU=";
+  };
+
+  vendorHash = "sha256-uQOmUXT2wd40DwwTCMnFFKd47eu+BPBDjiCGtUNFoKY=";
+
+  patchPhase = ''
+    sed -i -e '/func main()/,/^}/d' -e '/"io"/a "io/ioutil"' main.go
+    cat << EOF >> main.go
+    func main() {
+      outputFile, err := os.Create("geosite.db")
+      if err != nil {
+        panic(err)
+      }
+      defer outputFile.Close()
+      vData, err := ioutil.ReadFile("${v2ray-domain-list-community}/share/v2ray/geosite.dat")
+      if err != nil {
+        panic(err)
+      }
+      domainMap, err := parse(vData)
+      if err != nil {
+        panic(err)
+      }
+      err = geosite.Write(outputFile, domainMap)
+      if err != nil {
+        panic(err)
+      }
+    }
```

Comment:
You can use `substituteAll` and keep the replacement `main.go` in its own file.

Reply:
Do you mean puting the whole file in nixpkgs?

Reply:
Just the replacement (addition actually) part, a separate file is easier to read than heredoc.

Reply:
Done.